pkg/validation-webhook/snapshot.go

```diff
@@ -117,7 +169,30 @@ func decideSnapshot(snapshot, oldSnapshot *volumesnapshotv1beta1.VolumeSnapshot,
 	return reviewResponse
 }
 
-func decideSnapshotContent(snapcontent, oldSnapcontent *volumesnapshotv1beta1.VolumeSnapshotContent, isUpdate bool) *v1.AdmissionResponse {
+func decideSnapshotContentV1beta1(snapcontent, oldSnapcontent *volumesnapshotv1beta1.VolumeSnapshotContent, isUpdate bool) *v1.AdmissionResponse {
+	reviewResponse := &v1.AdmissionResponse{
+		Allowed: true,
+		Result:  &metav1.Status{},
+	}
+
+	if isUpdate {
+		// if it is an UPDATE and oldSnapcontent is valid, check immutable fields
```

Comment:
hmmm, this does not seem to be right.
there should be a logic similar to line 129.
if oldSnapshotContent is already invalid, the admission hook should allow the update.
while in "decideSnapshotContentV1", it does have that piece of logic, which is not needed

Reply:
hmm, looks like I removed the wrong block of code.  I meant to remove line 207.